types/crypto-hash/index.d.ts

```diff
@@ -0,0 +1,48 @@
+// Type definitions for crypto-hash 1.0
+// Project: https://github.com/sindresorhus/crypto-hash#readme
+// Definitions by: BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function sha1(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options?: OptionsHexOutput
+): Promise<string>;
+export function sha1(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options: OptionBufferOutput
+): Promise<ArrayBuffer>;
+
+export function sha256(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options?: OptionsHexOutput
+): Promise<string>;
+export function sha256(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options: OptionBufferOutput
+): Promise<ArrayBuffer>;
+
+export function sha384(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options?: OptionsHexOutput
+): Promise<string>;
+export function sha384(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options: OptionBufferOutput
+): Promise<ArrayBuffer>;
+
+export function sha512(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options?: OptionsHexOutput
+): Promise<string>;
+export function sha512(
+    buffer: string | ArrayBuffer | ArrayBufferView,
+    options: OptionBufferOutput
+): Promise<ArrayBuffer>;
+
+export interface OptionsHexOutput {
+    outputFormat?: 'hex';
```

Comment:
I'm probably missing something, but why is this `outputFormat` optional but the one below is not?

Reply:
Also, would it be simpler to use a union type, either at the parameter level or the property level?

Reply:
Because `hex` is the default value. And when `buffer` is specified, the return type changes. I'm open for suggestions on how to do it better.

Reply:
Oops, I missed the change in return type.  There's still no reason for `OptionsHexOutput.outputFormat` to be optional, is there?

Reply:
Well, it is optional, after all.

Reply:
I'm convinced.  Thanks!